lmwrapper/structs.py

```diff
@@ -46,6 +46,8 @@ class LmPrompt:
     This means it is possible that the default max might cause errors with long prompts.
     It recommended that you specify a limit yourself to have more predictable
     behaviour."""
+    max_completion_tokens: int | None = None
```

Comment:
actually thinking about this more, I think I prefer making the new arg be along the lines of `max_tokens_excluding_reasoning_tokens` for what openAI now uses for `max_tokens` and set what we call `max_tokens` be `max_completion_tokens`. Also since our total users is small, I'm also ok with the breaking change of just removing `max_tokens` from lmwrapper and only having `max_completion_tokens` and `max_completion_tokens_excluding_reasoning_tokens`. This naming seems clearer.

Reply:
lmk, if you have any preference or thoughts.

Reply:
The naming is confusing but it identically matches the OpenAI API. If we change the names, it will be unclear how our code relates to the OpenAI API without reading our code. Then people can take advantage of the already extensive discussion on the namespace confusion, e.g., https://community.openai.com/t/why-was-max-tokens-changed-to-max-completion-tokens/938077/5

Basically I think we should blame OpenAI for naming things confusingly but not make things even worse.

Reply:
hm.. But we have the opportunity to make it clearer than openai and avoid any user digging into whether they should choose `max_completion_tokens` or `max_tokens`. (since we are ok with breaking old use cases).

Though I realize I was misunderstanding things before, as from API there is no way to actually do `max_completion_tokens_excluding_reasoning_tokens` with the o1 models (I thought this would happen if you passed in `max_tokens`, but it just won't take that).

Ok, so I'm good with a version like this PR matching OAI just to get to this merged. Then in a future PR we just delete `max_tokens` and just have `max_completion_tokens` that maps depending on the model. There doesn't seem to be a good reason for both of these parameters to exist. As discussed in that thread you linked it really is weird design decision on OAI's part. A developer is already "opting in" into a new cost structure when they change the model. As far as I can tell there didn't need to be an arg name change.